docs/source/notes/get_start_xpu.rst

```diff
@@ -0,0 +1,417 @@
+
+Pytorch 2.4: Getting Started on Intel GPU
+=========================================
+
+The support for Intel GPUs is released alongside PyTorch v2.4.
+
+This release only supports build from source for Intel GPUs.
+
+Hardware Prerequiste
+--------------------
+
+.. list-table::
+   :header-rows: 1
+
+   * - Supported Hardware
+     - Intel® Data Center GPU Max Series
+   * - Supported OS
+     - Linux
+
+
+Pytorch for Intel GPUs is compatible with Intel® Data Center GPU Max Series and Intel® Core™ Ultra Processors, and only support OS Linux with release 2.4.
+
+Software Prerequiste
+--------------------
+
+Install driver and required packages by following the `PyTorch Installation Prerequisites for Intel GPUs <https://www.intel.com/content/www/us/en/developer/articles/tool/pytorch-prerequisites-for-intel-gpus.html>`_ to Get Started.
+
+Set up Environment
+------------------
+
+Before you begin, you need to set up the environment. This can be done by sourcing the setvars.sh script provided by the ``intel-for-pytorch-gpu-dev`` and  ``intel-pti-dev`` packages.
+
+.. code-block::
+
+   source ${ONEAPI_ROOT}/setvars.sh
+
+**Note**\ : the ONEAPI_ROOT is the folder you installed your ``intel-for-pytorch-gpu-dev`` and  ``intel-pti-dev`` packages, usually it is /opt/intel/oneapi/ or ~/intel/oneapi/.
+
+Build from source
+-----------------
+
+Now we have all the required packages installed and environment acitvated. Follow the following command to install pytorch, torchvision, torchaudio by building from source. Refer to official guide in `pytorch from source <https://github.com/pytorch/pytorch?tab=readme-ov-file#intel-gpu-support>`_\ , `vision from source <https://github.com/pytorch/vision/blob/main/CONTRIBUTING.md#development-installation>`_ and `audio from source <https://pytorch.org/audio/main/build.linux.html>`_.
+
+.. code-block::
+
+   # Get PyTorch Source Code
+   git clone --recursive https://github.com/pytorch/pytorch
+   cd pytorch
+   git checkout main # or checkout the specific release version >= v2.4
+   git submodule sync
+   git submodule update --init --recursive
+
+   # Get required packages for compilation
+   conda install cmake ninja
+   pip install -r requirements.txt
+
+   # Pytorch for Intel GPUs only support Linux platform for now.
+   # Install the required packages for pytorch compilation
+   conda install intel::mkl-static intel::mkl-include
```

Comment:
emmm. This is really unexpected! First, you noted above that we just installed all prerequisite environment and now, all of the sudden, user needs to install yet-few-other intel packages from **different** source. Secondly, isn't mkl included into the oneAPI set user installs as prerequisite? and isn't it pulled automatically by pytorch as submodule?

In any case, I should note that I never did this step myself and I was able to build pytorch with XPU support. Hence, I highly doubt this step is actually needed. I also should note that using conda environment  might be recommended, but should not be required. How to install these packages in non-conda environment?

Reply:
Hi @EikanWang @riverliuintel @jingxu10 , do we still need    conda install intel::mkl-static intel::mkl-include  in the compilation steps?

Reply:
Hi @dvrogozh , this mkl package is for PyTorch CPU optimization, not related to XPU, but it is a essential step in building from source part.

Reply:
@dvrogozh , these packages are required for CPU as the XPU wheel package supports both CPU and GPU.

Reply:
Consider my setup, please. I don't use conda and install for Ubuntu 22.04. I am following [1] to install oneAPI dependencies and intel drivers. Now, I build pytorch for XPU and it builds and works for me. I really do not understand why `conda install intel::mkl-static intel::mkl-include` step is needed! Moreover, to me doing this step directly contradicts some aspects of the setup I observe. The point is that it seems I do have MKL which I guess is coming as a dependency from `sudo apt install intel-for-pytorch-gpu-dev intel-pti-dev`. Thus a question - why do I need to install yet-another MKL thru conda? and whether there will be any conflicts?

And to add yet another dimension to all of that. Do you really using MKL in pytorch or you are using oneDNN. Or you are using both? These are different libraries:
* https://github.com/oneapi-src/oneDNN
* https://github.com/oneapi-src/oneMKL
Note that additional confusion here comes from some renames and branch out of onednn from onemkl which did happen in the past. So, https://github.com/intel/ideep/blob/master/.gitmodules pytorch dependency refers to  https://github.com/01org/mkl-dnn.git which actually redirects to https://github.com/oneapi-src/oneDNN

[1] https://www.intel.com/content/www/us/en/developer/articles/tool/pytorch-prerequisites-for-intel-gpus.html

And here are some snapshots from my setup. Take a look:
```
$ python setup.py develop
--   USE_BLAS              : 1
--     BLAS                : mkl
--     BLAS_HAS_SBGEMM     :
--   USE_LAPACK            : 1
--     LAPACK              : mkl
...
--   USE_MPS               : OFF
--   USE_MKL               : ON
--   USE_MKLDNN            : ON
--   USE_MKLDNN_ACL        : OFF
--   USE_MKLDNN_CBLAS      : OFF
...
--   Public Dependencies  : caffe2::mkl
--   Private Dependencies : Threads::Threads;pthreadpool;cpuinfo;pytorch_qnnpack;nnpack;XNNPACK;fbgemm;ittnotify;fp16;caffe2::openmp;tensorpipe;gloo;foxi_loader;rt;fmt::fmt-header-only;kineto;gcc_s;gcc;dl


$ dpkg -S /opt/intel/oneapi/mkl/
intel-oneapi-mkl-cluster-devel-2025.0, intel-oneapi-mkl-core-devel-common-2025.0, intel-oneapi-mkl-sycl-dft-2025.0, intel-oneapi-mkl-core-common-2025.0, intel-oneapi-mkl-sycl-include-2025.0, intel-oneapi-mkl-classic-include-2025.0, intel-oneapi-mkl-classic-include-common-2025.0, intel-oneapi-mkl-sycl-2025.0, intel-oneapi-mkl-sycl-data-fitting-2025.0, intel-oneapi-mkl-sycl-stats-2025.0, intel-oneapi-mkl-sycl-rng-2025.0, intel-oneapi-mkl-sycl-vm-2025.0, intel-oneapi-mkl-sycl-devel-common-2025.0, intel-oneapi-mkl-sycl-sparse-2025.0, intel-oneapi-mkl-sycl-devel-2025.0, intel-oneapi-mkl-core-devel-2025.0, intel-oneapi-mkl-sycl-lapack-2025.0, intel-oneapi-mkl-sycl-blas-2025.0, intel-oneapi-mkl-core-2025.0, intel-oneapi-mkl-cluster-devel-common-2025.0, intel-oneapi-mkl-cluster-2025.0: /opt/intel/oneapi/mkl

$ ls /opt/intel/oneapi/mkl/
2024.1  2025.0  latest

$ find /opt/ -iname "*mkl*"
### here goes a huge list of MKL includes, static and ashared libraries!
```

Reply:
@ZhaoqiongZ please make sure to address this ^^

Reply:
Hi @dvrogozh , this step will not influence xpu, but if skip it, the cpu pytorch performance will be influenced. So we have to add it and it is also added in the pytorch readme installation part.

Reply:
> this step will not influence xpu, but if skip it, the cpu pytorch performance will be influenced

You don't explain this in the comment before the install command. Instead comment reads "Install the required packages". For xpu they are not required at all as you just explained. Considering that this document is dedicated to XPU, that's confusing.

> So we have to add it

As I described in the example above I believe that I already have installed similar package which provide MKL dependency via `sudo apt install intel-for-pytorch-gpu-dev intel-pti-dev`. Why do I need to install same dependency again via different mechanism?

And, important, read thru my comment above - does pytorch really using mkl today?

Reply:
@dvrogozh The compilation doesn't only compile binaries for XPU, but also compile for CPU. We don't need to make it that clear which one is for which. For users, they just need to get a combo binary working on both XPU and CPU.
The `conda install intel:mklxxxx` is the default compilation step for PyTorch. We need to follow it.

Reply:
Hi @svekars , we discussed internally, pls merge at this time.

Reply:
I would like to emphasize, that I do not consider my comments addressed. And I preserve my vote on this PR: -1.